spec/unit/logstash_output_spec.rb

```diff
@@ -28,49 +33,26 @@
   end
 
   describe "plugin register" do
-    let(:registered_plugin) { plugin.tap(&:register) }
 
     describe "construct host URI" do
 
       it "applies default https scheme and 9800 port" do
-        expect(registered_plugin.construct_host_uri.eql?(::LogStash::Util::SafeURI.new("https://127.0.0.1:9800/"))).to be_truthy
+        expect(registered_plugin.send(:construct_host_uri).first.eql?("https://127.0.0.1:9800/")).to be_truthy
       end
 
       describe "SSL disabled" do
         let(:config) { super().merge("ssl_enabled" => false) }
 
         it "causes HTTP scheme" do
-          expect(registered_plugin.construct_host_uri.eql?(::LogStash::Util::SafeURI.new("http://127.0.0.1:9800/"))).to be_truthy
+          expect(registered_plugin.send(:construct_host_uri).first.eql?("http://127.0.0.1:9800/")).to be_truthy
         end
       end
 
       describe "custom port" do
         let(:config) { super().merge("hosts" => "127.0.0.1:9808") }
 
         it "will be applied" do
-          expect(registered_plugin.construct_host_uri.eql?(::LogStash::Util::SafeURI.new("https://127.0.0.1:9808/"))).to be_truthy
-        end
-      end
-    end
-
-    describe "username and password auth" do
-      let(:config) { super().merge("hosts" => "my-ls-downstream.com:1234", "ssl_enabled" => false) }
-
-      context "with `username`" do
-        let(:config) { super().merge("username" => "test_user") }
-
-        it "requires `password`" do
-          expected_message = "`password` is REQUIRED when `username` is provided"
-          expect{ plugin }.to raise_error(LogStash::ConfigurationError).with_message(expected_message)
-        end
-      end
-
-      context "with `password`" do
-        let(:config) { super().merge("password" => "pa$$") }
-
-        it "requires `username`" do
-          expected_message = "`password` not allowed unless `username` is configured"
-          expect{ registered_plugin }.to raise_error(LogStash::ConfigurationError).with_message(expected_message)
```

Comment:
is this behaviour no longer specified?

Reply:
Initially thought it is a duplicate of [http-mixin test](https://github.com/logstash-plugins/logstash-mixin-http_client/blob/main/spec/plugin_mixin/http_client_spec.rb#L44) but it is required now, especially after deprecating `user` and going with `username`.